### PR TITLE
jats modification to handle springer

### PR DESCRIPTION
**Pull request by mugdhapolimera**

### CODE REVIEW

File: adsingestp/parsers/jats.py [hunk 1]
```diff
@@ -190,24 +191,58 @@ def parse(self, article_metadata):
...
+            contribs_raw_init = []
+            if art_contrib_group.find_all("contrib", recursive=False):
+                contribs_raw_init = art_contrib_group.find_all("contrib", recursive=False)
+
             default_key = "ALLAUTH"
+
+            # cycle through <contrib> to check if a <collab> is listed in the same level as an author an has multiple authors nested under it; targeted for Springer
+            contribs_raw = []
+            collabflag = None
+            for i in range(len(contribs_raw_init)):
```
Comment (kelockhart) [inline adsingestp/parsers/jats.py:205]: You should be able to structure this as: `for contrib in contribs_raw_init`, which keeps you away from having to using indices everywhere. You do save off the index in `collabflag`, so you do need to access the index, but it's generally more common to write it this way: `for idx, contrib in enumerate(contribs_raw_init)` - then you use `contrib` to access the item itself, and only use `idx` when you specifically need to access the index.

File: adsingestp/parsers/jats.py [hunk 2]
```diff
@@ -190,24 +191,58 @@ def parse(self, article_metadata):
...
+                            ca.append(collabtag)
+                            contribs_raw.append(ca)
+                    else:
+                        contribs_raw.append(contribs_raw_init[i])
+
+                else:
+                    contribs_raw.append(contribs_raw_init[i])
+
             num_contribs = len(contribs_raw)
+            count = 0
```
Comment (kelockhart) [inline adsingestp/parsers/jats.py:226]: Is this used anywhere? If it's leftover from debugging, go ahead and remove

File: adsingestp/parsers/jats.py [hunk 3]
```diff
@@ -190,24 +191,58 @@ def parse(self, article_metadata):
...
+
             default_key = "ALLAUTH"
+
+            # cycle through <contrib> to check if a <collab> is listed in the same level as an author an has multiple authors nested under it; targeted for Springer
+            contribs_raw = []
+            collabflag = None
+            for i in range(len(contribs_raw_init)):
+                if contribs_raw_init[i].find("collab"):
+                    # find nested collab authors and unnest them
+                    if contribs_raw_init[i].find("collab").find("institution"):
```
Comment (kelockhart) [inline adsingestp/parsers/jats.py:208]: Add a comment here that the name of the collab is stored in the institution field - that took me a sec to figure out

File: adsingestp/parsers/jats.py [hunk 4]
```diff
@@ -190,24 +191,58 @@ def parse(self, article_metadata):
...
+                            contribs_raw.append(ca)
+                    else:
+                        contribs_raw.append(contribs_raw_init[i])
+
+                else:
+                    contribs_raw.append(contribs_raw_init[i])
+
             num_contribs = len(contribs_raw)
+            count = 0
             for contrib in contribs_raw:
```
Comment (kelockhart) [inline adsingestp/parsers/jats.py:227]: Ok, another thought on how to structure this and fold in the for loop up above with this one - what if you make this loop over `art_contrib_group.find_all("contrib", recursive=False)` - that'll get you all the top-level authors. Then, within the loop (at the top), do a `contrib.find("collab")` to see if the collab is at that author position. If so, do a `contrib.find_all("contrib")` to get the nested authors. You can add them to the top-level list of contribs like this:
```
for nested_contrib in nested_contribs:
        contrib.insert_after(nested_contrib.extract())
```
You'll need to do an insert_after for the collab itself too. That *should* make the nested collab authors get dealt with on the next loop iteration, but you'll have to double check that

**Review by kelockhart — CHANGES_REQUESTED** [review]
I had another idea for dealing with the nested collab author nonsense. The comment with that idea is chronologically after a bunch of smaller comments about your original implementation, so the order is probably confusing - let's chat tomorrow about it.

File: adsingestp/parsers/jats.py [hunk 5]
```diff
@@ -190,24 +191,58 @@ def parse(self, article_metadata):
...
+
+            # cycle through <contrib> to check if a <collab> is listed in the same level as an author an has multiple authors nested under it; targeted for Springer
+            contribs_raw = []
+            collabflag = None
+            for i in range(len(contribs_raw_init)):
+                if contribs_raw_init[i].find("collab"):
+                    # find nested collab authors and unnest them
+                    if contribs_raw_init[i].find("collab").find("institution"):
+                        # save the author position of collab to add collab as an author later
+                        collabflag = i
```
[the remark this answers is not in the document]
Comment (mugdhapolimera) [reply adsingestp/parsers/jats.py:210]: this is now inside a loop and it is set up to handle multiple unique collaborations

File: adsingestp/parsers/jats.py (hunk 1 above)
Comment (mugdhapolimera) [reply adsingestp/parsers/jats.py:205]: I took your advice and attempted a modification. It even caught some collabs that I missed the first time around. It works well now.

File: adsingestp/parsers/jats.py (hunk 4 above)
Comment (mugdhapolimera) [reply adsingestp/parsers/jats.py:227]: I took your advice and attempted a modification. It even caught some collabs that I missed the first time around. It works well now.

**Review by kelockhart — CHANGES_REQUESTED** [review]
Looks good! Just a couple small clean-up things left.

**Pull request merged**